### PR TITLE
QF-1784 Amend user journey for Gov UK One Login

### PR DESCRIPTION
**Pull request by karen-hanna**

### CODE REVIEW

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Contacts/CreateContactHandlerTests/CreateContactHandlerTests.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@ public async Task Handle_WhenContactDoesNotExist_ShouldCreateContact()
             // Assert
             result.Result.Should().BeTrue();
             _mockContactRepository.Verify(x => x.CreateNewContact(It.IsAny<Contact>()), Times.Once);
-            _mockContactRepository.Verify(x => x.UpdateSignInId(It.IsAny<Guid>(), It.IsAny<Guid>(), request.GovIdentifier), Times.Once);
+            _mockContactRepository.Verify(x => x.UpdateSignInId(It.IsAny<Guid>(), request.GovIdentifier), Times.Once);
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Contacts/CreateContactHandlerTests/CreateContactHandlerTests.cs:52]: SignInId renamed to UpdateGovUkIdentifier

File: src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Login/LoginHandlerTestsBase.cs [hunk 2]
```diff
@@ -25,10 +25,10 @@ public void Arrange()
             OrgQueryRepository = new Mock<IOrganisationQueryRepository>();
 
             ContactQueryRepository = new Mock<IContactQueryRepository>();
-            ContactQueryRepository.Setup(r => r.GetBySignInId(It.IsNotIn(Guid.Empty)))
+            ContactQueryRepository.Setup(r => r.GetContactFromGovIdentifier(It.IsNotIn(string.Empty)))
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Login/LoginHandlerTestsBase.cs:28]: GetContactFromGovidentifier renamed to GetContactByGovIdentifier

File: src/SFA.DAS.AssessorService.Web.UnitTests/AccountControllerTests/UpdateUserAccount.cs [hunk 3]
```diff
@@ -94,7 +94,7 @@ public async Task Then_If_Valid_Calls_Invite_User_With_Gov_Claims_Redirects_To_I
                 x.ValidateAsync(
                     It.Is<ValidationContext<AccountViewModel>>(c => c.InstanceToValidate.Equals(accountViewModel)),
                     CancellationToken.None), Times.Once);
-            Assert.AreEqual("InviteSent",actualViewResult.ActionName);
+            Assert.AreEqual("Index",actualViewResult.ActionName);
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AssessorService.Web.UnitTests/AccountControllerTests/UpdateUserAccount.cs:97]: It's no longer used.  I had already removed the methods, now the View has also been removed

File: src/SFA.DAS.AssessorService.Web/Controllers/AccountController.cs [hunk 4]
```diff
@@ -106,8 +106,7 @@ public async Task<IActionResult> PostSignIn()
                     _sessionService.Set("EndPointAssessorOrganisationId", epaoId);
                     return RedirectToAction("Rejected", "Home");
                 case LoginResult.ContactDoesNotExist:
-                    ResetCookies();
-                    return RedirectToAction("NotRegistered", "Home");
+                    return RedirectToAction("UpdateAnAccount", "Account");
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AssessorService.Web/Controllers/AccountController.cs:109]: Renamed to CreateAnAccount

File: src/SFA.DAS.AssessorService.Web/Controllers/AccountController.cs [hunk 5]
```diff
@@ -246,35 +245,13 @@ public async Task<IActionResult> UpdateAnAccount(AccountViewModel accountViewMod
...
+                await _contactsApiClient.InviteUser(new CreateContactRequest(accountViewModel.GivenName, accountViewModel.FamilyName, email, null, email, govIdentifier));
 
-            _sessionService.Set("NewAccount", JsonConvert.SerializeObject(new CreateAccountViewModel
-            {
-                Email = email,
-                FamilyName = accountViewModel.FamilyName,
-                GivenName = accountViewModel.GivenName
-            }));
-            return inviteSuccess.Result ? RedirectToAction("InviteSent") : RedirectToAction("Error", "Home");
+            return inviteSuccess.Result ? RedirectToAction("Index", "OrganisationSearch") : RedirectToAction("Error", "Home");
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AssessorService.Web/Controllers/AccountController.cs:252]: Yes.  View also removed

File: src/SFA.DAS.AssessorService.Web/Controllers/BaseController.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.AspNetCore.Authorization;
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AssessorService.Web/Controllers/BaseController.cs:1]: AssessorController has been merged into Base Controller.

File: src/SFA.DAS.AssessorService.Web/Controllers/AccountController.cs [hunk 7]
```diff
@@ -106,8 +106,7 @@ public async Task<IActionResult> PostSignIn()
                     _sessionService.Set("EndPointAssessorOrganisationId", epaoId);
                     return RedirectToAction("Rejected", "Home");
                 case LoginResult.ContactDoesNotExist:
-                    ResetCookies();
-                    return RedirectToAction("NotRegistered", "Home");
+                    return RedirectToAction("UpdateAnAccount", "Account");
                 default:
```
[the remark this answers is not in the document]
Comment (karen-hanna) [reply src/SFA.DAS.AssessorService.Web/Controllers/AccountController.cs:110]: Not Registered is still used in a number of places when the if an organisation can't be found with a given APIOrganisationId

File: src/SFA.DAS.AssessorService.Application.Api/Validators/DfeSignInCallbackValidator.cs [hunk 8]
```diff
@@ -9,9 +9,8 @@ public class DfeSignInCallbackValidator : AbstractValidator<SignInCallback>
     {
         public DfeSignInCallbackValidator(IStringLocalizer<CreateContactRequest> localizer)
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.AssessorService.Application.Api/Validators/DfeSignInCallbackValidator.cs:10]: Should rename this as GovUkSignInCallbackValidator - the Dfe bit was never true

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AssessorService.Application.Api/Validators/DfeSignInCallbackValidator.cs (hunk 8 above)
Comment (karen-hanna) [reply src/SFA.DAS.AssessorService.Application.Api/Validators/DfeSignInCallbackValidator.cs]: I have renamed it

**Pull request merged**